paddle/majel/place.h

```diff
@@ -14,33 +14,24 @@ struct CpuPlace {
 };
 
 struct GpuPlace {
-  GpuPlace(int d) : device(d) {}
+  GpuPlace() {}
 
   // needed for variant equality comparison
-  inline bool operator==(const GpuPlace& o) const { return device == o.device; }
+  inline bool operator==(const GpuPlace&) const { return true; }
 
-  inline bool operator!=(const GpuPlace& o) const { return !(*this == o); }
-
-  GpuPlace() : GpuPlace(0) {}
-  int device;
```

Comment:
We can run a neural network on one device, but when we aggregate gradients/parameters from these devices, I think we need to copy data from/to devices. Thus we need to be able to specify the exact place as the source/destination.